src/services/item/plugins/app/appData/repository.ts

```diff
@@ -1,65 +1,60 @@
 
-import { AppDataVisibility, ItemType, Member, PermissionLevel, UUID } from '@graasp/sdk';
+import { AppDataVisibility, FileItemType, ItemType, Member, PermissionLevel } from '@graasp/sdk';
 
-import { AppDataSource } from '../../../../../plugins/datasource';
-import { MemberIdentifierNotFound } from '../../../../itemLogin/errors';
+import { MutableRepository } from '../../../../../repositories/MutableRepository';
+import { DEFAULT_PRIMARY_KEY } from '../../../../../repositories/const';
+import { IllegalArgumentException } from '../../../../../repositories/errors';
 import { appDataSchema } from '../../../../member/plugins/export-data/schemas/schemas';
 import { schemaToSelectMapper } from '../../../../member/plugins/export-data/utils/selection.utils';
 import { AppData, Filters } from './appData';
 import { AppDataNotFound, PreventUpdateAppDataFile } from './errors';
 import { InputAppData } from './interfaces/app-data';
 
-export const AppDataRepository = AppDataSource.getRepository(AppData).extend({
-  async post(itemId: string, actorId: Member['id'], body: Partial<InputAppData>): Promise<AppData> {
-    const created = await this.insert({
-      ...body,
-      item: { id: itemId },
-      creator: { id: actorId },
-      member: { id: body.memberId ?? actorId },
-    });
-
-    // TODO: better solution?
-    // query builder returns creator as id and extra as string
-    return this.get(created.identifiers[0].id);
-  },
+type CreateAppDataBody = { appData: Partial<InputAppData>; itemId: string; actorId: Member['id'] };
+type UpdateAppDataBody = DeepPartial<InputAppData>;
 
-  async patch(itemId: string, appDataId: string, body: Partial<AppData>): Promise<AppData> {
-    // shouldn't update file data
-    // TODO: optimize and refactor
-    const originalData = await this.get(appDataId);
-    if ([ItemType.LOCAL_FILE, ItemType.S3_FILE].includes(originalData?.data?.type)) {
-      throw new PreventUpdateAppDataFile(originalData.id);
-    }
+const RELATIONS = { member: true, creator: true, item: true };
 
-    await this.update({ id: appDataId, item: { id: itemId } }, body);
+export class AppDataRepository extends MutableRepository<AppData, UpdateAppDataBody> {
+  constructor(manager?: EntityManager) {
+    super(DEFAULT_PRIMARY_KEY, AppData, manager);
+  }
 
-    // TODO: optimize
-    return this.get(appDataId);
-  },
+  async addOne({ itemId, actorId, appData }: CreateAppDataBody): Promise<AppData> {
+    return await super.insert(
+      {
+        ...appData,
+        item: { id: itemId },
+        creator: { id: actorId },
+        member: { id: appData.memberId ?? actorId },
+      },
+      RELATIONS,
+    );
+  }
 
-  async deleteOne(itemId: string, appDataId: string): Promise<UUID> {
-    const deleteResult = await this.delete(appDataId);
+  async updateOne(appDataId: string, body: UpdateAppDataBody): Promise<AppData> {
+    // shouldn't update file data
+    // TODO: optimize and refactor
```

Comment:
We should probably create an Issue instead of having a todo here ?

Reply:
Yeah I agree, it was already here ^^

Reply:
Ah my bad then ! I though you added it 😅

Reply:
Yes I agree, I just moved the code, it's why there is still the TODO here ^^
I think I'll remove the TODO, because if we want to prevent updating app data file, we can't do it without doing a select before...